src/go/serviceconfig/service_config_fetcher.go

```diff
@@ -15,239 +15,74 @@
+// service configs, pick up the one with highest traffic percentage.
+func (s *ServiceConfigFetcher) LoadConfigIdFromRollouts() (string, error) {
+	rollouts := new(smpb.ListServiceRolloutsResponse)
+	fetchRolloutUrl := util.FetchRolloutsURL(s.serviceManagementUrl, s.serviceName)
+	if err := util.CallGooglelapis(s.client, fetchRolloutUrl, util.GET, s.accessToken, rollouts); err != nil {
+		return "", err
 	}
 
-	return scf.callServiceManagementRollouts(util.FetchRolloutsURL(scf.opts.ServiceManagementURL, scf.serviceName), token)
+	return highestTrafficConfigIdInLatestRollout(rollouts)
 }
 
-// TODO(taoxuy): replace this with callServiceControl for configId
-func (scf *ServiceConfigFetcher) callServiceManagementRollouts(path, token string) (*smpb.ListServiceRolloutsResponse, error) {
-	var err error
-	var resp *http.Response
-	if resp, err = scf.callWithAccessToken(path, token); err != nil {
-		return nil, err
+func highestTrafficConfigIdInLatestRollout(rollouts *smpb.ListServiceRolloutsResponse) (string, error) {
+	if rollouts == nil || len(rollouts.GetRollouts()) == 0 {
+		return "", fmt.Errorf("problematic rollouts: %v", rollouts)
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("fail to read response body: %s", err)
-	}
-	defer resp.Body.Close()
-	rolloutsResponse := new(smpb.ListServiceRolloutsResponse)
-	if err := proto.Unmarshal(body, rolloutsResponse); err != nil {
-		return nil, fmt.Errorf("fail to unmarshal ListServiceRolloutsResponse: %s", err)
-	}
-	return rolloutsResponse, nil
-}
+	latestRollout := rollouts.GetRollouts()[0]
 
-func (scf *ServiceConfigFetcher) callServiceManagement(path, token string) (*confpb.Service, error) {
-	var err error
-	var resp *http.Response
-	if resp, err = scf.callWithAccessToken(path, token); err != nil {
-		return nil, err
-	}
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("fail to read response body: %s", err)
-	}
-	defer resp.Body.Close()
-	service := new(confpb.Service)
-	if err := proto.Unmarshal(body, service); err != nil {
-		return nil, fmt.Errorf("fail to unmarshal Service: %v", err)
+	highestPercent := 0.
+	highTrafficConfigId := ""
+	for configId, percent := range latestRollout.GetTrafficPercentStrategy().Percentages {
+		if percent > highestPercent {
+			highestPercent = percent
+			highTrafficConfigId = configId
+		}
 	}
-	return service, nil
-}
 
-func (scf *ServiceConfigFetcher) callWithAccessToken(path, token string) (*http.Response, error) {
-	req, _ := http.NewRequest("GET", path, nil)
-	req.Header.Add("Authorization", "Bearer "+token)
-	req.Header.Set("Content-Type", "application/x-protobuf")
-	resp, err := scf.client.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	if resp.StatusCode != http.StatusOK {
-		resp.Body.Close()
-		return nil, fmt.Errorf("http call to %s returns not 200 OK: %v", path, resp.Status)
+	if !(math.Abs(100.0-highestPercent) < 1e-9) {
```

Comment:
Is this needed? We don't care about the absolute percentage, just the highest one.

Reply:
This is just for warning no other service configs except the highest one are not applied

Reply:
This is not easy to read,  maybe just 

highestPercent < 100.0